paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc

```diff
@@ -701,17 +701,41 @@ void ExecuteMatMulV2(const ExecutionContext &ctx,
   auto &astream = MKLDNNDeviceContext::tls().get_stream();
   matmul_p->execute(astream, matmul_args);
   astream.wait();
-
-  auto format = paddle::platform::MKLDNNFormatForSize(
-      out->dims().size(), dnnl::memory::format_tag::nchw);
-  out->set_layout(paddle::framework::DataLayout::kMKLDNN);
+  auto format =
+      MKLDNNFormatForSize(out->dims().size(), dnnl::memory::format_tag::nchw);
   out->set_format(format);
+  out->set_layout(DataLayout::kMKLDNN);
 }
 
 template <typename T>
 class MatMulV2MKLDNNKernel : public paddle::framework::OpKernel<T> {
  public:
-  void Compute(const ExecutionContext &ctx) const override { RunKernel(ctx); }
+  void Compute(const ExecutionContext &ctx) const override {
+    if (ctx.HasAttr("head_number")) {
```

Comment:
matmul_v2 doesn't have the attribute `head_number` defined, so this `if` statement will always be false

Reply:
Yes matmul_v2 doesn't have all those attributes, but I register matmul to the same kernel as matmul_v2 https://github.com/PaddlePaddle/Paddle/blob/754feed0b8743c94947bf801a7766b64b9e21ebc/paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc#L1261. That is why I check everywhere if attr exists. 
MatMulMKLDNNKernel I will remove in the next PR because it is still used by matmul_grad.

Reply:
Oh, I have overlooked that. Thanks for an explanation!